data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py

```diff
@@ -1,131 +1,200 @@
-
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99)
+"""
+Apply value ranges to ensure that values are reasonable and to minimize the likelihood
+of sensitive information (like phone numbers) within the free text fields.
 
-UNION ALL
+Original Issues: DC-827, DC-502, DC-487
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99)
-
-UNION ALL
+The intent is to ensure that numeric free-text fields that are not manipulated by de-id
+have value range restrictions applied to the value_as_number field across the entire dataset.
+"""
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99)
+# Python imports
+import logging
 
-UNION ALL
+# Third party imports
+from jinja2 import Environment
 
-SELECT
-  *
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.bq_utils import WRITE_TRUNCATE
+
+LOGGER = logging.getLogger(__name__)
+
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+# Query to create tables in sandbox with the rows that will be removed per cleaning rule
+SANDBOX_QUERY = jinja_env.from_string("""
+CREATE OR REPLACE TABLE
+    `{{project}}.{{sandbox_dataset}}.{{intermediary_table}}` AS (
+SELECT *
 FROM
-  `{project}.{dataset}.observation`
+    `{{project}}.{{dataset}}.observation`
 WHERE
-  observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99)
-
-UNION ALL
-
-SELECT
-  *
+    (observation_concept_id = 1585889 AND (value_as_number < 0 OR value_as_number > 20))
+OR
+    (observation_concept_id = 1585890 AND (value_as_number < 0 OR value_as_number > 20))
+OR
+    (observation_concept_id = 1585795 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1585802 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255))
+OR
+    (observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99))
+OR 
+    (observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1586162 AND (value_as_number < 0 OR value_as_number > 99)))
```

Comment:
Looks good to merge.

Suggestion, These conditions can also be simplified lite the other query in this cleaning rule.
```
(observation_concept_id IN (1585889, 1585890) AND (value_as_number < 0 OR value_as_number > 20)) OR
    (observation_concept_id IN (1585795, 1585802, 1585864, 1585870, 1585873, 1586159, 1586162) AND (value_as_number < 0 OR value_as_number > 99)) OR
    (observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255))

```